src/progress/mod.rs

```diff
@@ -195,7 +197,14 @@ impl Progress {
     /// Returns false if the given index comes from an out of order message.
     /// Otherwise it decreases the progress next index to min(rejected, last)
     /// and returns true.
-    pub fn maybe_decr_to(&mut self, rejected: u64, last: u64, request_snapshot: u64) -> bool {
+    pub fn maybe_decr_to<T: Storage>(
```

Comment:
This looks ugly to me. `maybe_decr_to` intends to decrease replicate index to min(rejected, last). Can it be figured outside the function?